packages/web/vite.config.ts

```diff
@@ -1,10 +1,29 @@
 import path from 'path';
 import { defineConfig } from 'vite';
 import react from '@vitejs/plugin-react-swc';
+import chokidar from 'chokidar';
 
 // https://vitejs.dev/config/
 export default defineConfig({
-  plugins: [react()],
+  plugins: [
+    react(),
+    // reload on backend change
+    {
```

Comment:
Why do we need this on the client, to reload the websocket connection?

Reply:
this should just be reloading the ws on the client whenever the API rebuilds in dev mode which is needed otherwise it's connected to a nonexistent session and breaks but @swk777 correct me if i'm wrong on that

Reply:
this should just be reloading the ws on the client whenever the API rebuilds in dev mode which is needed otherwise it's connected to a nonexistent session and breaks but @swk777 correct me if i'm wrong on that

Reply:
That's what I assume. I would be ok with this as a stop-gap but the better solution here is for the client side websocket client to re-establish connections when the connection is closed. Robust websocket clients use a combination of heartbeats and onclose listeners to attempt reconnect with a backoff strategy.

For now I'm ok to merge this but I would like to comment and/or file an issue to capture this.

Reply:
There is already some retry logic inside of the client, but I think this is retrying to send a message if it fails, so maybe it doesn't handle this case properly.

https://github.com/srcbookdev/srcbook/blob/main/packages/web/src/clients/websocket/client.ts#L10-L11

Reply:
https://github.com/srcbookdev/srcbook/issues/280

Reply:
@versecafe is right, and as @benjreinhart mentioned, this approach is indeed better. I’ll work on implementing this later. 
Another point that deserves attention is that in the current design, the cellId is randomly generated, which changes after a reload. As a result, cell operations cannot be continued seamlessly across reloads.

Reply:
That's a good point, we need to fix those.